### ISSUE TITLE
ui: gracefully handle 500 responses on PHP errors

### ISSUE THREAD

**Issue by fichtner** [opnsense/core#5879]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Not sure how and why this started but now we get empty 500 responses on PHP errors at least for static pages. It might be better to print some boilerplate 500 page with the error embedded so it can be reported more easily.

**To Reproduce**

Add error to 22.7 static PHP page. Get 500 empty response.

**Expected behavior**

See a page with the error at least.

**Comment by kulikov-a** [comment]
Hi. isn't this "on purpose" (debug vs. production mode?) ;)
https://github.com/opnsense/core/blob/04cc8489dea88f1f53a1900a7e3585cd563de56a/src/opnsense/service/templates/OPNsense/WebGui/php.ini#L28

**Comment by fichtner** [comment]
@kulikov-a Aha, this should at least include development as well then. But I’m not sure what gain this has on production to be silent about errors (actual errors only now)

**Comment by kulikov-a** [comment]
>  not sure what gain this has on production to be silent about errors (actual errors only now)

can redirect 500 errors to `/crash_reporter.php` on lighty for "product" mode? (but I don't see much benefit either)

**Comment by fichtner** [comment]
Actually that might be a nice thing to do: redirect to crash reporter page in case of 500 error? It’ll display the error in any case.

**Comment by kulikov-a** [comment]
nothing smarter than https://github.com/kulikov-a/core/commit/6e3ed08812a51dfdeb79526a91a0613eddaa3d44 from me )

**Comment by fichtner** [comment]
@kulikov-a thanks I committed slightly adjusted version: we redirect on production/development but not on debug where display_errors is on. I'd really like to see (fatal) errors rendered in the page for development but display_errors has the nag to display ALL "errors" as defined by the error level which makes turning on display_errors useless as pages could break like they did in the past writing errors to HTML.

**Comment by fichtner** [comment]
@kulikov-a ok this was worse than expected... did some other things to avoid breaking API 500 returns ;(

**Comment by kulikov-a** [comment]
@fichtner yes, I tried it and it seems to be what needed . the only question is do we still need a static  `status-500.html` in this case :)

**Comment by fichtner** [comment]
nope, gone :)
